helm-charts/basehub/values.schema.yaml

```diff
@@ -283,25 +283,21 @@ properties:
             required:
               - templateVars
```

Comment:
gitRepoUrl and gitRepoBranch is also required for this to work and can be added here.

Reply:
I agree. But isn't this obsolete since we're setting defaults in the basehub chart? I noticed that the config set by basehub is not listed in the schema and I guess it's to avoid the toil of checking every single bit of config.
But I'll add these two since they're new.

Actually, I will also add `jupyterhub.ingress` and `jupyterhub.proxy` to the schema, as we might miss to add it for future hubs since they've always been in basehub.

Reply:
Done in https://github.com/2i2c-org/infrastructure/pull/2349/commits/3f5a15ed46a255091f3270da2021bed34ee2e504

Reply:
We depend on them to not error, so they are just as required as `templateVars`. Practically, since we have templateVars as required, we are already catching a common where one unset all values in `custom.homepage` for example - then its good that at least one variable is required because otherwise the schema validation would pass.

I'd like us to work towards being consistently thorough in the schema, so that we and anyone else can use the schema to understand what is and what is and isn't required by our templates etc. I've made z2jh and binderhub consistently thorough with things like this and it has helped for sure overall, for example catching a few "oops we unset all default values by setting a dictionary to null" situations.

Reply:
> Actually, I will also add jupyterhub.ingress and jupyterhub.proxy to the schema, as we might miss to add it for future hubs since they've always been in basehub.

They are already handled by the jupyterhub schema and they are marked as required by the jupyterhub schema already.

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/6d3c13984d78791f6c6980a70ddf8cf0648cf723/jupyterhub/schema.yaml#L17-L22

You could make the schema tighter though, for example requiring a non-empty value for some fields etc. For example currently `proxy.hosts` is [allowed to be a blank array](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/6d3c13984d78791f6c6980a70ddf8cf0648cf723/jupyterhub/schema.yaml#L1742-L1751), while in basehub we require it to be set to something for a functional setup.

Reply:
@consideRatio, does this look ok https://github.com/2i2c-org/infrastructure/pull/2349/commits/89f445ee704747a5a0a1cedecf04c1e71ce64719?

Reply:
Already caught an issue in utoronto: https://github.com/2i2c-org/infrastructure/actions/runs/4511216409/jobs/7943109492 yay 🎉

Reply:
Wiee nice yepp this looks great!!!